### PR TITLE
Adjust `DifficultyValue` curve to avoid lower star rating of osu!taiko being too inflated

### PR DESCRIPTION
**Pull request by YaniFR**
This pull request aims to change the actual DifficultyValue Curve to a Splitted Curve Calculation to ensure a lower DifficultyValue on lower StarRating (under 5.5*) without affecting higher StarRating (above 5.5) 

You can see the actual graph of the modified curve 

And the difference between the value where g(x) equal the new curve

Here are some example :

cosMo@BousouP feat. Hatsune Miku - Hatsune Miku no Shoushitsu (Chromoxx) [Hivie's Muzukashii] 5.06* for SS : 
Confirmed Changes : 293pp 
Curve Adjustment : 281pp

Chroma - sink to the deep sea world (Idealism) [hivie's ura muzukashii] 4.45* for SS : 
Confirmed Changes : 246pp 
Curve Adjustment : 225pp

### CODE REVIEW

File: osu.Game.Rulesets.Taiko/Difficulty/TaikoPerformanceCalculator.cs [hunk 1]
```diff
@@ -73,7 +73,9 @@ protected override PerformanceAttributes CreatePerformanceAttributes(ScoreInfo s
 
         private double computeDifficultyValue(ScoreInfo score, TaikoDifficultyAttributes attributes)
         {
-            double difficultyValue = Math.Pow(5 * Math.Max(1.0, attributes.StarRating / 0.115) - 4.0, 2.25) / 1150.0;
+            double baseFormula = 5 * Math.Max(1.0, attributes.StarRating / 0.115) - 4.0;
+
```
Comment (Lawtrohux) [inline osu.Game.Rulesets.Taiko/Difficulty/TaikoPerformanceCalculator.cs:77]:
```suggestion
            double difficultyValue = Math.Min(Math.Pow(baseFormula, 3) / 69052.51, Math.Pow(baseFormula, 2.25) / 1150.0);
```

File: osu.Game.Rulesets.Taiko/Difficulty/TaikoPerformanceCalculator.cs [hunk 2]
```diff
@@ -73,7 +73,9 @@ protected override PerformanceAttributes CreatePerformanceAttributes(ScoreInfo s
 
         private double computeDifficultyValue(ScoreInfo score, TaikoDifficultyAttributes attributes)
         {
-            double difficultyValue = Math.Pow(5 * Math.Max(1.0, attributes.StarRating / 0.115) - 4.0, 2.25) / 1150.0;
+            double baseFormula = 5 * Math.Max(1.0, attributes.StarRating / 0.115) - 4.0;
+            
+            double difficultyValue = Math.Min(Math.Pow(baseFormula, 3) / 69052.51, Math.Pow(baseFormula, 2.25) /1150.0);
```
Comment (Lawtrohux) [inline osu.Game.Rulesets.Taiko/Difficulty/TaikoPerformanceCalculator.cs:78]:
```suggestion
```

**Review by Lawtrohux — CHANGES_REQUESTED** [review]
formatting needs to be fixed

**Review by Lawtrohux — APPROVED** [review]
Lgtm. Whilst there may be some apprehensions on magic numbers used, scaling and graph both make sense.

**Comment by smoogipoo** [comment]
!diffcalc
RULESET=taiko

**Comment by Lawtrohux** [comment]
Works as intended, still lgtm.

**Review by stanriders — APPROVED** [review]
Reapproving after a master merge

**Pull request merged**